src/components/vega-lite/index.tsx

```diff
@@ -50,31 +55,69 @@ export class VegaLite extends React.PureComponent<VegaLiteProps, any> {
     //   }
     // };
     const {logger} = this.props;
+    const vlSpec = this.props.spec;
     try {
       const spec = vl.compile(vlSpec, logger).spec;
       const runtime = vega.parse(spec, vlSpec.config);
-      const view = new vega.View(runtime)
+      this.view = new vega.View(runtime)
         .logLevel(vega.Warn)
         .initialize(this.refs[CHART_REF] as any)
         .renderer(this.props.renderer || 'svg')
-        .hover()
-        .run();
-      vegaTooltip.vega(view);
+        .hover();
+      vegaTooltip.vega(this.view);
+      this.bindData();
     } catch (err) {
       logger.error(err);
     }
   }
 
   protected componentDidMount() {
-    this.renderVega(this.props.spec);
+    this.updateSpec();
+    this.runView();
+  }
+
+  protected componentWillUpdate() {
+    this.size = this.getChartSize();
```

Comment:
Can you explain why you have to set size after componentWillUpdate?  

Note that the size of the plot can depend on data, which means that we have to read the size from SVG and set the size to the surrounding container _after_ `view.run()`.

Reply:
Oh, but this is shadow DOM, so maybe you're right.  I'm not sure.  If it works after `view.run()`, I would put it here though.

Reply:
We want to make the plot size persist when clicking the sort button, right? Getting size in componentWillUpdate will give us the previous plot size, and the new plot should be this size.